tests/test_halo_data_transformer.py

```diff
@@ -0,0 +1,471 @@
+    )
+
+    specification = HaloUpdateSpec(
+        n_halo_points=n_halos,
+        shape=quantity.data.shape,
+        strides=quantity.data.strides,
+        itemsize=quantity.data.itemsize,
+        origin=quantity.metadata.origin,
+        extent=quantity.metadata.extent,
+        dims=quantity.metadata.dims,
+        numpy_module=quantity.np,
+        dtype=quantity.metadata.dtype,
+    )
+
+    exchange_descriptors = [
+        HaloExchangeData(specification, boundary_north, 0, boundary_north),
+        HaloExchangeData(specification, boundary_southwest, 0, boundary_southwest),
+    ]
+
+    data_transformer = HaloDataTransformer.get(quantity.np, exchange_descriptors)
+
+    assert len(data_transformer.get_pack_buffer().array.shape) == 1
+    assert (
+        data_transformer.get_pack_buffer().array.size
+        == quantity.data[boundary_north].size + quantity.data[boundary_southwest].size
+    )
+    assert len(data_transformer.get_unpack_buffer().array.shape) == 1
+    assert (
+        data_transformer.get_unpack_buffer().array.size
+        == quantity.data[boundary_north].size + quantity.data[boundary_southwest].size
+    )
+    # clean up
+    Buffer.push_to_cache(data_transformer._pack_buffer)
+    Buffer.push_to_cache(data_transformer._unpack_buffer)
+
+
+def _get_boundaries(quantity, n_halos):
+    send_boundaries = {}
+    recv_boundaries = {}
+    for direction in [
+        NORTH,
+        NORTHWEST,
+        WEST,
+        SOUTHWEST,
+        SOUTH,
+        SOUTHEAST,
+        EAST,
+        NORTHEAST,
+    ]:
+        send_boundaries[direction] = _boundary_utils.get_boundary_slice(
+            quantity.dims,
+            quantity.origin,
+            quantity.extent,
+            quantity.data.shape,
+            direction,
+            n_halos,
+            interior=True,
+        )
+        recv_boundaries[direction] = _boundary_utils.get_boundary_slice(
+            quantity.dims,
+            quantity.origin,
+            quantity.extent,
+            quantity.data.shape,
+            direction,
+            n_halos,
+            interior=False,
+        )
+
+    return send_boundaries, recv_boundaries
+
+
+def test_data_transformer_scalar_pack_unpack(quantity, rotation, n_halos):
+    original_quantity: Quantity = copy.deepcopy(quantity)
+
+    send_boundaries, recv_boundaries = _get_boundaries(quantity, n_halos)
+    N_edge_boundaries = {
+        0: (send_boundaries[NORTH], recv_boundaries[SOUTH]),
+        -1: (send_boundaries[NORTH], recv_boundaries[WEST]),
+        -2: (send_boundaries[NORTH], recv_boundaries[NORTH]),
+        -3: (send_boundaries[NORTH], recv_boundaries[EAST]),
+    }
+
+    NE_corner_boundaries = {
+        0: (send_boundaries[NORTHEAST], recv_boundaries[SOUTHEAST]),
+        -1: (send_boundaries[NORTHEAST], recv_boundaries[SOUTHWEST]),
+        -2: (send_boundaries[NORTHEAST], recv_boundaries[NORTHWEST]),
+        -3: (send_boundaries[NORTHEAST], recv_boundaries[NORTHEAST]),
+    }
```

Comment:
nit on a nit: can you switch the order these are defined in so I don't initially read this as "number of edge boundaries"?

Reply:
Done